grabber/main.go

```diff
@@ -142,72 +142,96 @@ func getLatestBlockNumber(importer *backend.Backend) int64 {
 	}
 	return number
 }
+
+// backfill continuously loops over all blocks in reverse order, verifying that all
+// data is loaded and consistent, and reloading as necessary.
 func backfill(importer *backend.Backend, startFrom int64) {
 	blockNumber := startFrom
 	if startFrom <= 0 {
 		blockNumber = getLatestBlockNumber(importer)
 	}
+	var hash common.Hash // Expected hash.
```

Comment:
Tracking the parent hash along the way allows dropping the call to `checkParentForBlock()`, which should also eliminate some redundant calls.